library/cwm/src/lib/cwm/widget.rb

```diff
@@ -0,0 +1,450 @@
+require "yast"
+
+require "abstract_method"
+
+module CWM
+  # Represent base for any widget used in CWM. It can be passed as "widget" argument. For more
+  # details about usage see {CWM.ShowAndRun}
+  class AbstractWidget
+    include Yast::UIShortcuts
+    include Yast::I18n
+
+    # @return [String] id used for widget
+    attr_accessor :widget_id
+
+    # defines widget type for CWM usage
+    def self.widget_type=(val)
```

Comment:
I don't get it. Why not simply define an instance method `widget_type` in the children of AbstractWidget, returning the appropriate string?